Microsoft.Toolkit.Uwp.UI.Controls/DockPanel/DockPanel.cs

```diff
@@ -0,0 +1,135 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using Microsoft.Toolkit.Uwp.UI.Extensions;
+using Windows.Foundation;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// Defines an area where you can arrange child elements either horizontally or vertically, relative to each other.
+    /// </summary>
+    public partial class DockPanel : Panel
+    {
+        private static void DockChanged(DependencyObject sender, DependencyPropertyChangedEventArgs e)
+        {
+            var senderElement = sender as FrameworkElement;
+            var dockPanel = senderElement?.FindParent<DockPanel>();
+
+            if (dockPanel == null)
+            {
+                return;
+            }
+
+            dockPanel.InvalidateArrange();
+            dockPanel.InvalidateArrange();
+        }
+
+        private static void LastChildFillChanged(DependencyObject sender, DependencyPropertyChangedEventArgs e)
+        {
+            var dockPanel = (DockPanel)sender;
+            dockPanel.InvalidateMeasure();
+            dockPanel.InvalidateArrange();
```

Comment:
There's no need to invalidate arrange. Invalidating measure invalidates arrange too